src/test/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/ProcessMessageServiceTest.java

```diff
@@ -0,0 +1,276 @@
+
+        givenHmcStatusUpdateCaseDataWillReturnSscsCaseDetails(sscsCaseDetails, CANCELLED);
+
+        // when
+        processMessageService.processEventMessage(hmcMessage);
+
+        // then
+        verify(caseStateUpdateService, times(1))
+                .updateCancelled(hearingGetResponse, caseData);
+        verifyUpdateCaseDataCalledCorrectlyForHmcStatus(caseData, CANCELLED);
+    }
+
+    @DisplayName("When no cancellation reason is given but status is Cancelled, "
+            + "updateCancelled and updateCaseData are not called")
+    @Test
+    void testUpdateCancelledNullReason() throws Exception {
+        // given
+        hearingGetResponse.getRequestDetails().setStatus("Cancelled");
+        hmcMessage.getHearingUpdate().setHmcStatus(CANCELLED);
+
+        givenHmcStatusUpdateCaseDataWillReturnSscsCaseDetails(sscsCaseDetails, CANCELLED);
+
+        // when
+        processMessageService.processEventMessage(hmcMessage);
+
+        // then
+        verify(caseStateUpdateService, times(1))
+                .updateCancelled(hearingGetResponse, caseData);
+        verifyUpdateCaseDataCalledCorrectlyForHmcStatus(caseData, CANCELLED);
+
+    }
+
+    @DisplayName("When invalid status given in request and no cancellation reason is given,"
+            + "updateCancelled and updateCaseData are not called")
+    @ParameterizedTest
+    @ValueSource(strings = {"test"})
+    @NullAndEmptySource
+    void testUpdateCancelledInvalidStatusNullReason(String value) throws Exception {
+        // given
+        hearingGetResponse.getRequestDetails().setStatus(value);
+        hmcMessage.getHearingUpdate().setHmcStatus(CANCELLED);
+
+        // when
+        processMessageService.processEventMessage(hmcMessage);
+
+        // then
+        verify(caseStateUpdateService, never()).updateCancelled(any(), any());
+        verify(ccdCaseService, never()).updateCaseData(any(),any(),any(),any());
+    }
+
+    @DisplayName("When HmcStatus is Exception updateFailed and updateCaseData are called")
+    @Test
+    void testUpdateCancelledInvalidStatusNullReason() throws Exception {
+        // given
+        hmcMessage.getHearingUpdate().setHmcStatus(EXCEPTION);
+
+        givenHmcStatusUpdateCaseDataWillReturnSscsCaseDetails(sscsCaseDetails, EXCEPTION);
+
+        // when
+        processMessageService.processEventMessage(hmcMessage);
+
+        // then
+        verify(caseStateUpdateService, times(1))
+                .updateFailed(caseData);
+        verifyUpdateCaseDataCalledCorrectlyForHmcStatus(caseData, EXCEPTION);
+    }
+
+    @DisplayName("When not listed, updated, canceled or exception nothing is called")
+    @Test
+    void testInvalidCall() throws Exception {
+        // given
+        hmcMessage.getHearingUpdate().setHmcStatus(null);
+
+        // when
+        processMessageService.processEventMessage(hmcMessage);
+
+        // then
+        verify(caseStateUpdateService, never()).updateListed(any(), any(), any());
```

Comment:
minor/optional: again `verifyZeroInteractions` for this? like how this is explicit however could be a pain to maintain.

Reply:
Done where possible